pandas/tests/indexes/datetimes/test_arithmetic.py

```diff
@@ -44,7 +45,74 @@ def addend(request):
 
 
 class TestDatetimeIndexComparisons(object):
-    # TODO: De-duplicate with test_comparisons_nat below
+    @pytest.mark.parametrize('other', [datetime(2016, 1, 1),
+                                       Timestamp('2016-01-01'),
+                                       np.datetime64('2016-01-01')])
+    def test_dti_cmp_datetimelike(self, other, tz):
+        dti = pd.date_range('2016-01-01', periods=2, tz=tz)
+        if tz is not None:
+            if isinstance(other, np.datetime64):
+                # no tzaware version available
+                return
+            elif isinstance(other, Timestamp):
+                other = other.tz_localize(dti.tzinfo)
+            else:
+                other = tslib._localize_pydatetime(other, dti.tzinfo)
+
+        result = dti == other
+        expected = np.array([True, False])
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = dti > other
+        expected = np.array([False, True])
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = dti >= other
+        expected = np.array([True, True])
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = dti < other
+        expected = np.array([False, False])
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = dti <= other
+        expected = np.array([True, False])
+        tm.assert_numpy_array_equal(result, expected)
+
+    def dti_cmp_non_datetime(self, tz):
+        # GH#19301 by convention datetime.date is not considered comparable
+        # to Timestamp or DatetimeIndex.  This may change in the future.
+        dti = pd.date_range('2016-01-01', periods=2, tz=tz)
+
+        other = datetime(2016, 1, 1).date()
+        assert not (dti == other).any()
+        assert (dti != other).all()
+        with pytest.raises(TypeError):
+            dti < other
+        with pytest.raises(TypeError):
+            dti <= other
+        with pytest.raises(TypeError):
+            dti > other
+        with pytest.raises(TypeError):
+            dti >= other
+
+    @pytest.mark.parametrize('other', [None,
+                                       np.nan])
```

Comment:
can you add ``pd.NaT`` here

Reply:
No, pd.NaT behaves differently from None or np.nan.

Reply:
huh? that would be troubling, why is that

Reply:
The `__eq__` and `__ne__` behave the same, but None and np.nan raise `TypeError` for the inequalities, whereas pd.NaT returns False.

Reply:
right. ok then split the tests to reflect this, testing all nulls at once for eq/ne make the test easy to grok.

Reply:
> make the test easy to grok.

I'll do this, but object to the notions that fixtures and easy-to-grok can go in the same file.  If I can't import the test and run it in the interpreter, then I'm in a Strange Land.